compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -2146,15 +2147,19 @@ public void markParameterizedType(BType enclosingType, Collection<BType> constit
 
     // private methods
 
-    private BSymbol resolveOperator(ScopeEntry entry, List<BType> typeList) {
+    private BSymbol resolveOperator(boolean isAdditiveExpr, ScopeEntry entry, List<BType> typeList) {
         BSymbol foundSymbol = symTable.notFoundSymbol;
         while (entry != NOT_FOUND_ENTRY) {
             BInvokableType opType = (BInvokableType) entry.symbol.type;
             if (typeList.size() == opType.paramTypes.size()) {
                 boolean match = true;
                 for (int i = 0; i < typeList.size(); i++) {
                     BType t = Types.getReferredType(typeList.get(i));
-                    if ((t.getKind() == TypeKind.UNION) &&
+                    if (isAdditiveExpr) {
```

Comment:
Why do we need to special case additive expression here?

Reply:
Changed the logic